src/client/Microsoft.Identity.Client/Platforms/Features/WinFormsLegacyWebUi/WindowsFormsWebAuthenticationDialogBase.cs

```diff
@@ -262,7 +271,7 @@ internal AuthorizationResult AuthenticateAAD(Uri requestUri, Uri callbackUri)
             // The WebBrowser event handlers must not throw exceptions.
             // If they do then they may be swallowed by the native
             // browser com control.
-            _webBrowser.Navigating += WebBrowserNavigatingHandler;
+            _webBrowser.BeforeNavigate += WebBrowserBeforeNavigateHandler;
```

Comment:
What is the difference between the 2 events?

Reply:
Using `BeforeNavigate2` instead of `Navigating` because it exposes post data. But `BeforeNavigate2` doesn't have a default .NET wrapper, so similar to `NavigateError`, have to create our own wrapper, event handlers, event args.